src/plot.rs

```diff
@@ -23,51 +36,167 @@ pub fn plot_total_energy(data: Vec<(f64, f64)>, config: &Config) {
+    let mut y_min_potential_energy = data
         .iter()
-        .map(|e| e.0)
+        .map(|e| e.potential_energy)
         .fold(f64::INFINITY, |a, b| a.min(b));
-    let x_max = data
+    let mut _y_max_potential_energy = data
         .iter()
-        .map(|e| e.0)
+        .map(|e| e.potential_energy)
         .fold(-f64::INFINITY, |a, b| a.max(b));
-    let y_min = data
+    let mut y_min_kinetic_energy = data
         .iter()
-        .map(|e| e.1)
+        .map(|e| e.kinetic_energy)
         .fold(f64::INFINITY, |a, b| a.min(b));
-    let y_max = data
+    let mut _y_max_kinetic_energy = data
         .iter()
-        .map(|e| e.1)
+        .map(|e| e.kinetic_energy)
         .fold(-f64::INFINITY, |a, b| a.max(b));
+    // add 5% padding around the max and min values
+    x_min *= 0.95;
+    x_max *= 1.05;
+    _y_min_total_energy *= 0.95;
+    y_max_total_energy *= 1.05;
+    y_min_potential_energy *= 0.95;
+    _y_max_potential_energy *= 1.05;
+    y_min_kinetic_energy *= 0.95;
+    _y_max_kinetic_energy *= 1.05;
+    let min_energy = min(y_min_potential_energy, y_min_kinetic_energy);
+
+    let y_label_size = root_drawing_area
+        .estimate_text_size(
+            &format_label(&y_max_total_energy),
+            &TextStyle {
+                font: ("Sans-serif", 20).into_font(),
+                color: BLACK.to_backend_color(),
+                pos: Pos {
+                    h_pos: plotters::style::text_anchor::HPos::Center,
+                    v_pos: plotters::style::text_anchor::VPos::Center,
+                },
+            },
+        )
+        .expect("Should be able to estimate the text size");
 
-    let mut chart = ChartBuilder::on(&root_drawing_area)
-        .caption("Energy", ("Sans-serif", 20).into_font())
-        .x_label_area_size(20)
-        .y_label_area_size(40)
-        .build_cartesian_2d(x_min..x_max, y_min..y_max)
-        .unwrap();
+    let x_label_size = root_drawing_area
+        .estimate_text_size(
+            &format_label(&y_max_total_energy),
+            &TextStyle {
+                font: ("Sans-serif", 15).into_font(),
+                color: BLACK.to_backend_color(),
+                pos: Pos {
+                    h_pos: plotters::style::text_anchor::HPos::Center,
+                    v_pos: plotters::style::text_anchor::VPos::Center,
+                },
+            },
+        )
+        .expect("Should be able to estimate the text size");
+    let mut chart_context = match ChartBuilder::on(&root_drawing_area)
+        .caption("System Energy over Time", ("Sans-serif", 20).into_font())
+        .x_label_area_size(x_label_size.0)
+        .y_label_area_size(y_label_size.0)
+        .build_cartesian_2d(x_min..x_max, min_energy..y_max_total_energy)
+    {
+        Ok(chart) => chart,
+        Err(e) => {
+            tracing::event!(tracing::Level::ERROR, "Error while creating chart: {e}");
+            return;
+        }
+    };
 
-    chart
+    match chart_context
```

Comment:
I wonder if we could use anyhow to change our function signature to a generic error so that we can get away with the ?-operator:
```
chart_context.(...)?;
```

Reply:
We probably can 👍🏻 I'll do that.